pkg/kudoctl/packages/types.go

```diff
@@ -10,6 +10,22 @@ const (
 	APIVersion = "kudo.dev/v1beta1"
 )
 
+// Resolver will try to resolve a given package name to either local tarball, folder, remote url or
+// an operator in the remote repository.
+type Resolver interface {
+	Resolve(name string, appVersion string, operatorVersion string) (*FancyResources, error)
+}
+
+// FancyResources (naming is hard :=) type provides operator resources together with a "scope-aware" dependency resolver.
+// For example, a dependency like `./child-operator` in an local directory needs the host system to be resolved, but when
+// found in a tarball, it needs the contents of the tarball because the dependency is not allowed to "escape" into the
+// host system.
+type FancyResources struct {
```

Comment:
How about naming this `PackageScope` as it scopes the resources, path and means to resolve package-specific dependencies?

Reply:
Yeah... That's not too bad. Maybe `PackageWithScope` or `PackageWithContext` or `PackageContext` would work as well.